src/include/processor/operator/scan/scan_rel_table.h

```diff
@@ -10,13 +11,23 @@ struct ScanRelTableInfo {
     storage::RelTable* table;
     common::RelDataDirection direction;
     std::vector<common::column_id_t> columnIDs;
+    std::vector<storage::ColumnPredicateSet> columnPredicates;
+
+    std::unique_ptr<storage::RelTableScanState> localScanState;
```

Comment:
Can u add a TODO here that we should move `localScanState` out of info struct eventually? Same for scan node table.